### PR TITLE
DT-6446 New analytics for Waltti

### PR DESCRIPTION
**Pull request by sharhio**
## Proposed Changes

  - Configurable analytics script
  - Configured for Waltti and excluded for Varely as it uses waltti config but is not under waltti
  - Hostname taken from request to distinguish between different aliases of the same instances (fe. reittiopas.hameenlinna vs hameenlinna.digitransit)
  - Dev and test environments excluded

## Pull Request Check List

  - A reasonable set of unit tests is included
  - Console does not show new warnings/errors
  - Changes are documented or they are self explanatory
  - This pull request does not have any merge conflicts
  - All existing tests pass in CI build

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform
  - Check that the implementation matches the design, when such one is defined in a Jira issue
  - Merge the pull request

### CODE REVIEW

File: app/configurations/config.varely.js [hunk 1]
```diff
@@ -170,4 +170,5 @@ export default configMerger(walttiConfig, {
   showNearYouButtons: true,
   allowLogin: false,
   routeNotifications: [],
+  analyticsScript: '',
```
Comment (vesameskanen) [inline app/configurations/config.varely.js:173]: maybe not needed? Empty string vs function in waltti config.

File: app/util/analyticsUtils.js [hunk 2]
```diff
@@ -35,16 +35,20 @@ export function addAnalyticsEvent(event) {
  *
  * @return string
  */
-export function getAnalyticsInitCode(GTMid) {
-  if (!GTMid) {
-    return '';
+export function getAnalyticsInitCode(GTMid, analyticsScript, hostname) {
```
Comment (vesameskanen) [inline app/util/analyticsUtils.js]: I suggest changing this to getAnalyticsInitCode(config, hostname)

**Review by vesameskanen — CHANGES_REQUESTED** [review]

File: app/configurations/config.varely.js (hunk 1 above)
Comment (sharhio) [reply app/configurations/config.varely.js:173]: The config is needed here to stop the analytics script from being added to varely instance from waltti config. Empty string is the minimalist way to stop it but I was wondering if it's a little bit confusing.

Comment (partisaani) [reply app/configurations/config.varely.js:173]: Especially since the field is called a script, would a void function `() => {}` be less confusing?

Comment (vesameskanen) [reply app/configurations/config.varely.js:173]: Empty string works as javascript thinks it is a 'falsy' value.  I guess we can accept this.

**Review by vesameskanen — APPROVED** [review]

**Pull request merged**